### ISSUE TITLE
Wrong query on altering column to a non-nullable datatype

### ISSUE THREAD

**Issue by obirider** [jOOQ/jOOQ#6470]
When executing this statement 

```
context.alterTable("Appointment").alter("closing_time").set(SQLDataType.TIMESTAMP.nullable(false)).execute();
```

I get the following exception:

```
Caused by: org.jooq.exception.DataAccessException: SQL [alter table "Appointment" alter "closing_time" type timestamp not null]; ERROR: syntax error at or near "not"
  Position: 63
        at org.jooq_3.9.4.POSTGRES_9_4.debug(Unknown Source)
        at org.jooq.impl.Tools.translate(Tools.java:1994)
        at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:676)
        at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:363)
        ...
        ... 67 more
Caused by: org.postgresql.util.PSQLException: ERROR: syntax error at or near "not"
  Position: 63
        at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2476)
        at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2189)
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:300)
        at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:428)
        at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:354)
        at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:169)
        at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:158)
        at org.jooq.tools.jdbc.DefaultPreparedStatement.execute(DefaultPreparedStatement.java:195)
        at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:431)
        at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:349)
       ... 69 more
```

**Comment by lukaseder** [comment]
Thank you very much for your report. This is a known issue: #3805. I'll close this as a duplicate. We can certainly consider a fix for PostgreSQL in 3.10, though.

Emulating changing the type *and* the constraint in one go can be done as such:

```sql
-- "Normal" statement
ALTER TABLE "t" ALTER "col" TYPE INT NOT NULL;

-- PostgreSQL
DO $$
BEGIN
  ALTER TABLE "t" ALTER "col" TYPE INT;
  ALTER TABLE "t" ALTER "col" SET NOT NULL;
END$$;
```

**Comment by lukaseder** [comment]
#3805 is fixed in jOOQ 3.10. It depends on new API, which is why we cannot merge this to 3.9 and older branches. jOOQ 3.10 is expected to be released by the end of Q3 2017.

Thanks again for reporting.
